packages/block/src/block.ts

```diff
@@ -49,15 +49,18 @@ export class Block {
     // parse uncle headers
     const uncleHeaders = []
     for (const uhData of uhsData ?? []) {
+      const cachedFork = header._common.hardfork()
       const uh = BlockHeader.fromHeaderData(uhData, {
-        ...opts,
+        hardforkByBlockNumber: true,
+        ...opts, // This potentially overwrites hardforkByBlocknumber
         // Use header common in case of hardforkByBlockNumber being activated
         common: header._common,
```

Comment:
if we do `common: header._common.copy()`, then could we skip needing to reset back to the cachedFork?

Reply:
Hmm, `Common` is now copied on the structurally static blockchain objects (so: block and tx) anyhow on instantiation, see [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/block/src/header.ts#L203), so I guess this shouldn't make a difference? 🤔

Reply:
Oh that's funny, I did not notice it. I'll try removing it, to see if it works.

Reply:
Yep, this indeed works. Ready for re-review.